airflow/jobs.py

```diff
@@ -398,6 +403,7 @@ def schedule_dag(self, dag):
                 )
                 session.add(next_run)
                 session.commit()
```

Comment:
@mistercrunch I'm not sure I understand the question. Apart from not having removed any calls to `expunge_all`, my understanding is that `commit` should leave the session in a clean and safe state.